_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp

```diff
@@ -848,12 +848,21 @@ namespace UMC_MPEG2_DECODER
         return frame;
     }
 
-    static
-    bool IsFieldOfOneFrame(const MPEG2PictureHeader& picHdr, const MPEG2PictureCodingExtension& picExtHdr,
-                           const MPEG2PictureHeader & newPictureHdr, const MPEG2PictureCodingExtension& newPicExtHdr)
+    bool MPEG2Decoder::IsFieldOfSameFrame() const
     {
-        if (picHdr.temporal_reference != newPictureHdr.temporal_reference)  // 6.3.9
-            return false;
+        const auto firstFrameSlice = m_currFrame->GetAU(0)->GetSlice(0);
+        const auto picHdr = firstFrameSlice->GetPicHeader();
+        const auto picExtHdr = firstFrameSlice->GetPicExtHeader();
+        const auto newPicHdr = *m_currHeaders.picHdr.get();
+        const auto newPicExtHdr = *m_currHeaders.picExtHdr.get();
+
+        // for I-frame we don't need to check if temporal refs are eq
```

Comment:
need to highlighting that this is WA.  "for I-frame we don't need to check" - this is wrong, we need to check!

Reply:
ok